recipes/dd-handler.rb

```diff
@@ -46,11 +46,20 @@
   ENV['DATADOG_PROXY'] = proxy_url.to_s
 end
 
+extra_endpoints = []
+node['datadog']['extra_endpoints'].each do |_, endpoint|
+  next unless endpoint['enabled']
+  endpoint.delete('enabled')
```

Comment:
This will attempt to mutate `node['datadog']['extra_endpoints']` too, which chef doesn't allow from a recipe (it'll cause a compile error).

I think we should either copy the `endpoint` hash first (keeping in mind that it's not a regular hash but an [`ImmutableMash`](http://www.rubydoc.info/github/opscode/chef/Chef/Node/ImmutableMash) if I remember correctly) or "manually" assign the interesting fields of `endpoint` to a new hash.

Also, unless the resulting values of `extra_endpoints` var are [Mash](http://www.rubydoc.info/gems/chef/Mash)es I think their keys need to be string literals (`:url` instead of `'url'`) since that's how we access the values in the handler.